.github/workflows/release.yml

```diff
@@ -0,0 +1,148 @@
+## (5b) Pin to commits, not tags
+
+name: CI Build And Publish Release
+
+on:
+  release:
+    types:
+      - published
+
+permissions:
+  contents: read
+
+env:
+  IMAGE: rancher/kube-api-auth
+
+jobs:
+  push:
+    permissions:
+      contents: read
+      id-token: write # this is important, it's how we authenticate with Vault
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+          - linux/arm64
+    steps:
+      - name: Prepare Matrix Instance
+        run: |
+          platform=${{ matrix.platform }}
+          echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
+
+      - name: Retrieve Docker Hub Credentials From Vault
+        uses: rancher-eio/read-vault-secrets@main
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKERHUB_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKERHUB_PASSWORD
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ env.DOCKERHUB_USERNAME }}
+          password: ${{ env.DOCKERHUB_PASSWORD }}
+
+      - name: Get Sources
+        uses: actions/checkout@v4
+
+      - name: Docker Meta Data Setup
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.IMAGE }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          cache: false
+
+      - name: Build and stage
+        run: |
+          case "${{ matrix.platform }}" in
+          */arm64) export ARCH=arm64 ; export GOARCH=arm64 ;;
+          *)       export ARCH=amd64 ; export GOARCH=amd64 ;;
+          esac
```

Comment:
We can use the same multi-dimensional matrix approach to avoid this `case`.

Reply:
Will do.

Reply:
Done with commit 301d15954ab974155391551b86e71f15bf5884c6